panic: concurrent write to websocket connection

Using the latest commit to the `master`this happens. I have also confirmed it present on the `0.1.2` version.

```
panic: concurrent write to websocket connection

goroutine 88169580 [running]:
github.com/gorilla/websocket.(*messageWriter).flushFrame(0xc011df6d20, 0xc009445901, 0x0, 0x0, 0x0, 0x7f12f5486e10, 0x0)
	/home/solidity/go/pkg/mod/github.com/gorilla/websocket@v1.4.1/conn.go:610 +0x612
github.com/gorilla/websocket.(*messageWriter).Close(0xc011df6d20, 0x8, 0xc053db3d00)
	/home/solidity/go/pkg/mod/github.com/gorilla/websocket@v1.4.1/conn.go:724 +0x62
github.com/gorilla/websocket.(*Conn).beginMessage(0xc02b0aa2c0, 0xc0094459b0, 0x2, 0x442401, 0xc053db3ee8)
	/home/solidity/go/pkg/mod/github.com/gorilla/websocket@v1.4.1/conn.go:473 +0x284
github.com/gorilla/websocket.(*Conn).NextWriter(0xc02b0aa2c0, 0x2, 0x0, 0x0, 0xa55, 0x100000000000000)
	/home/solidity/go/pkg/mod/github.com/gorilla/websocket@v1.4.1/conn.go:513 +0x53
github.com/gorilla/websocket.(*Conn).WriteMessage(0xc02b0aa2c0, 0x2, 0xc0463d7b00, 0x20, 0x40, 0x10, 0xc04d8f44e0)
	/home/solidity/go/pkg/mod/github.com/gorilla/websocket@v1.4.1/conn.go:766 +0x73
github.com/libp2p/go-ws-transport.(*Conn).Write(0xc011df69c0, 0xc0463d7b00, 0x20, 0x40, 0x1178340, 0x12eee40, 0x7f12f00d9320)
	/home/solidity/go/pkg/mod/github.com/libp2p/go-ws-transport@v0.1.3-0.20191008032742-3098bba549e8/conn_native.go:70 +0x59
bytes.(*Buffer).WriteTo(0xc009445980, 0x7f12a01a9480, 0xc011df6cf0, 0x7f12f00d9320, 0xc009445980, 0xcd4101)
	/usr/local/go/src/bytes/buffer.go:239 +0xb5
io.copyBuffer(0x7f12a01a9480, 0xc011df6cf0, 0x16068a0, 0xc009445980, 0x0, 0x0, 0x0, 0x8, 0x439416, 0x14a6618)
	/usr/local/go/src/io/io.go:384 +0x33f
io.Copy(...)
	/usr/local/go/src/io/io.go:364
github.com/multiformats/go-multistream.SelectProtoOrFail.func1(0x132712c, 0xa, 0x7f12a01a93e8, 0xc011df6cf0, 0xc04a438480)
	/home/solidity/go/pkg/mod/github.com/multiformats/go-multistream@v0.1.0/client.go:27 +0x163
created by github.com/multiformats/go-multistream.SelectProtoOrFail
	/home/solidity/go/pkg/mod/github.com/multiformats/go-multistream@v0.1.0/client.go:23 +0x95
```

I previously opened this on the `go-multistream` repo https://github.com/multiformats/go-multistream/issues/49. Failure is happening [here](https://github.com/libp2p/go-ws-transport/blob/master/conn_native.go#L70). I suspect we need to protect the connection by a mutex?

--- a/conn_native.go
+++ b/conn_native.go
@@ -17,6 +17,7 @@ type Conn struct {
 	DefaultMessageType int
 	reader             io.Reader
 	closeOnce          sync.Once
+	mx                 sync.Mutex
 }
 
 func (c *Conn) Read(b []byte) (int, error) {
@@ -67,6 +68,9 @@ func (c *Conn) prepNextReader() error {
 }
 
 func (c *Conn) Write(b []byte) (n int, err error) {
+	c.mx.Lock()
+	defer c.mx.Unlock()
+
 	if err := c.Conn.WriteMessage(c.DefaultMessageType, b); err != nil {
 		return 0, err
 	}
@@ -78,6 +82,9 @@ func (c *Conn) Write(b []byte) (n int, err error) {
 // close error, subsequent and concurrent calls will return nil.
 // This method is thread-safe.
 func (c *Conn) Close() error {
+	c.mx.Lock()
+	defer c.mx.Unlock()
+
 	var err error
 	c.closeOnce.Do(func() {
 		err1 := c.Conn.WriteControl(

